### PR TITLE
SorrIssue342 Misc improvements to the on boarding flow

### PR DESCRIPTION
**Pull request by samarth9008**
1. Added information about the build environment to check on the new interns.
2. Added info about our philosophy and what we expect from interns.

### CODE REVIEW

File: docs/How_to_organize_your_work.md [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
 - I'll remind you an adage that I tell myself when my teammates drive me crazy:
   "If you want to go fast, go alone; if you want to go far, go together"
+- While being assigned a warm-up issue:
+  - We value the importance of following instructions for efficient task
+    completion. We specifically pick the most straightforward simple issues
+    as the first ones so we’re not questioning whether you find a
+    solution for the task (most of the time we even give you interfaces, so
+    should not be a big deal). We want to see if you can follow simple rules
+    because no matter how smart you are until you learn how to follow the
+    instructions, you waste more of our time than save
```
Comment (DanilYachmenev) [inline docs/How_to_organize_your_work.md:16]: lets rephrase for a bit more correct phrasing

```
- We highly value the importance of following instructions to ensure efficient task completion. We intentionally select straightforward and simple issues as the first tasks for you. So as your first step you should put primary focus on following the rules. Failure to do so can result in wasting more of our time than saving it.
```

File: docs/Quick_start_for_developing.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
 <!--ts-->
```
Comment (DanilYachmenev) [inline docs/Quick_start_for_developing.md:1]: it seems like TOC here is generated not by `lint_md`
could you fix it pls?

File: docs/Quick_start_for_developing.md [hunk 3]
```diff
@@ -113,25 +121,43 @@
...
+
+  ```
+  alias sp='echo '\''source ~/.profile'\''; source ~/.profile'
+  alias vi='/usr/bin/vi'
+  alias vim='/usr/bin/vi'
+  alias vimdiff='/usr/bin/vi -d'
+  alias vip='vim -c "source ~/.vimrc_priv"'
+  alias w='which'
+  ==> SUCCESS <==
+  ```
```
Comment (DanilYachmenev) [inline docs/Quick_start_for_developing.md:147]: let's [indent](https://github.com/sorrentum/sorrentum/blob/master/docs/Documentation_about_guidelines.md#indenting-code-style) code blocks right under the bullet points they are related to

**Review by DanilYachmenev — COMMENTED** [review]
great! couple moments and ready

**Review by DanilYachmenev — COMMENTED** [review]
gj, lgtm!

**Review by gpsaggese — COMMENTED** [review]
Good work

**Pull request merged**